services/ui-src/src/components/reports/ReportPageFooter.tsx

```diff
@@ -32,17 +33,18 @@ export const ReportPageFooter = ({
           {!form?.id || formIsDisabled ? (
             <Button
               onClick={() => navigate(nextRoute)}
-              rightIcon={<Icon icon="arrowRight" />}
+              rightIcon={loading ? <></> : <Icon icon="arrowRight" />}
```

Comment:
Nit: I'd probably go with `undefined` here instead of forcing it to a fragment. So like:

```suggestion
              rightIcon={loading ? undefined : <Icon icon="arrowRight" />}
```
or
```suggestion
              rightIcon={!loading ? <Icon icon="arrowRight" /> : undefined}
```